src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -430,6 +441,23 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
                 new BufferedReader(new InputStreamReader(partitionInputStream)),
                 CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
         iterator = parser.iterator();
+
+        try {
+          if (!iterator.hasNext()) {
```

Comment:
No, sorry, I just picked a really bad example earlier. This ```hasNext()``` case has nothing to do with quotes. I meant the case where a worker _only_ has a partial record ending with a new line. Our logic would read this partial record up to the new line, but then have nothing else to read after. That is the case I'm trying to cover.

In other words, ```hasNext()``` simply makes sure that there is something to read after we've parsed up to the new line.

I was required to add the try/catch because this ```hasNext()``` check could blow up if we have a row with a truncated quote.